### ISSUE TITLE
[BEHAVIOR-1K / π0.5] Gripper never closes although navigation and arm reaching work correctly

### ISSUE THREAD

**Issue by xieshuaiyang02** [Physical-Intelligence/openpi#1012]
Hi, I am reproducing the official π0.5 baseline for the BEHAVIOR-1K 2026 Challenge with the R1Pro robot, and I encountered a consistent gripper issue.

Both the officially released checkpoint and my own fine-tuned checkpoint show essentially the same behavior:

The robot can correctly identify/find the target object.
It can navigate to the correct location.
The arm can reach toward the target object correctly.
However, the robot never executes the grasp / gripper-closing action.

For example, on turning_on_radio, the robot successfully navigates to the radio and reaches it with the arm, but the gripper remains open and the grasp is never triggered.

I printed the final 23-D whole-body actions sent by the policy during the entire rollout. The gripper dimensions remain close to +1 throughout:

[B1K ACTION] left_gripper[14]=+0.9983, right_gripper[22]=+0.9999
[B1K ACTION] left_gripper[14]=+0.9967, right_gripper[22]=+1.0019
[B1K ACTION] left_gripper[14]=+1.0009, right_gripper[22]=+1.0016
[B1K ACTION] left_gripper[14]=+0.9945, right_gripper[22]=+0.9981
...

This behavior continues even when the end effector has already reached the target object. I never observe a transition toward a negative gripper command corresponding to closing.

My setup is:

Model: π0.5
Robot: b1k/R1Pro
Model action horizon: 32
Execution horizon: 16
Whole-body action dimension: 23
Control mode: receding_horizon

I also checked whether this could be caused by missing or highly imbalanced gripper supervision in the training data.

For the raw turning_on_radio demonstrations, I obtained:

LEFT gripper:
close (-1): 32.44%
open  (+1): 67.56%

RIGHT gripper:
close (-1): 29.90%
open  (+1): 70.10%

So grasp actions are not particularly rare in the demonstrations.

I then inspected the actions after the complete B1K data preprocessing and normalization pipeline, i.e. the actual targets received by the model.

The normalization statistics are:

actions dim 14:
mean = 0.33794856
std  = 0.94116455

actions dim 22:
mean = 0.40144923
std  = 0.91588128

The resulting normalized gripper targets are:

LEFT:
OPEN  = +0.7034
CLOSE = -1.4216

RIGHT:
OPEN  = +0.6535
CLOSE = -1.5302

Sampling 25,600 transformed action targets gives:

LEFT dim14:
OPEN:  64.918%
CLOSE: 35.082%

RIGHT dim22:
OPEN:  71.438%
CLOSE: 28.562%

Therefore, the gripper-close targets appear to be preserved correctly by the training data pipeline.

One detail I am unsure about is the R1Pro representation. The physical robot has 25 DoF, while the policy uses a 23-D action representation, where the two finger joints of each gripper are represented by a single gripper action dimension (14 and 22). The state preprocessing similarly combines the two finger positions into a single gripper-width value.

Could there be a mismatch in the gripper state/action preprocessing, normalization, or inference-time postprocessing between the released checkpoint and the current BEHAVIOR/OpenPI pipeline?

Since the official checkpoint and my independently fine-tuned checkpoint both exhibit the same failure mode, while navigation and arm reaching work correctly, I suspect this may be related specifically to the gripper representation or inference pipeline rather than general policy training.

Has anyone observed this behavior with the official turning_on_radio π0.5 checkpoint?

Any suggestions on which intermediate gripper outputs or preprocessing stages I should inspect would be greatly appreciated.

**Comment by KhanhNgoDuy** [comment]
Hi, how did you fix this bug?

**Comment by xieshuaiyang02** [comment]
the model is loser only.no problem.haha

---- 回复的原邮件 ----
| 发件人 | Ngô Duy ***@***.***> |
| 发送日期 | 2026年08月24日 16:17 |
| 收件人 | Physical-Intelligence/openpi ***@***.***> |
| 抄送人 | xieshuaiyang ***@***.***>,
State ***@***.***> |
| 主题 | Re: [Physical-Intelligence/openpi] [BEHAVIOR-1K / π0.5] Gripper never closes although navigation and arm reaching work correctly (Issue #1012) |
KhanhNgoDuy left a comment (Physical-Intelligence/openpi#1012)

Hi, how did you fix this bug?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by KhanhNgoDuy** [comment]
So you changed the policy and it worked?

**Comment by xieshuaiyang02** [comment]
No change. I find it success finally. About 5% rate——turn on the radio task.

---- 回复的原邮件 ----
| 发件人 | Ngô Duy ***@***.***> |
| 发送日期 | 2026年08月24日 17:54 |
| 收件人 | Physical-Intelligence/openpi ***@***.***> |
| 抄送人 | xieshuaiyang ***@***.***>,
State ***@***.***> |
| 主题 | Re: [Physical-Intelligence/openpi] [BEHAVIOR-1K / π0.5] Gripper never closes although navigation and arm reaching work correctly (Issue #1012) |
KhanhNgoDuy left a comment (Physical-Intelligence/openpi#1012)

So you changed the policy and it worked?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!
